### PR TITLE
Migrate more Java packages to dev-java/log4j-12-api

### PR DESCRIPTION
**Pull request by Leo3418**
A patch for `net-p2p/vuze` was added to make it compatible with Log4j 2, which could save it from a last rite otherwise (https://github.com/gentoo/gentoo/pull/24245).  As a side effect, more memory is required by the JVM running `javac` during build time.  An extra amount of 192 MB memory was needed on my `amd64` test machines, and I assumed the same amount for other architectures.

`app-text/jabref` and `net-p2p/vuze` ebuilds were also reworked for EAPI 8, compliance with XDG conventions, and so on.

Please let me know if the change to `dev-java/ant-core` is trivial enough to warrant an immediate clean-up of `-r2`.

### CODE REVIEW

File: app-text/jabref/jabref-2.10-r7.ebuild [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	dev-java/spin:0
+	dev-java/microba:0
+	>=dev-java/glazedlists-1.8.0:0"
+
+DEPEND="
+	virtual/jdk:1.8
+	${CP_DEPEND}"
+
+RDEPEND="
+	virtual/jre:1.8
```
Comment (fordfrog) [inline app-text/jabref/jabref-2.10-r7.ebuild:40]: please add a comment why we need to restrict to `:1.8`

**Comment by fordfrog** [comment]
please squash the commits for each ebuild into one.

**Comment by fordfrog** [comment]
`dev-java/ant-core` merged, thanks! i dropped `-r2`, but just because this is a security revbump and we want to get stabilized this one and get rid of the old ones. otherwise i'd keep the old one as it would be soon eligible for stabilization.

File: app-text/jabref/jabref-2.10-r7.ebuild (hunk 1 above)
Comment (Leo3418) [reply app-text/jabref/jabref-2.10-r7.ebuild:51]: done!

**Comment by fordfrog** [comment]
`app-text/jabref` merged, thanks! i just dropped that >1.8 note from the commit message as we stick to `:1.8`.

**Comment by fordfrog** [comment]
`net-p2p/vuze` merged too, thanks!

**Comment by Leo3418** [comment]
> `app-text/jabref` merged, thanks! i just dropped that >1.8 note from the commit message as we stick to `:1.8`.

That note actually just refers to JDK in `DEPEND`, as the JDK atom in `DEPEND` is the `>=virtual/jdk-1.8:*` wildcard. Only the JRE version in `RDEPEND` is limited to `1.8`. But since the commit is already merged, let's keep it as-is.

**Pull request closed without merging**